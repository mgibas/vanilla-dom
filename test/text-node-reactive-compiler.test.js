const compiler = require('../src/text-node-reactive-compiler')

describe('text-node-reactive-compiler', () => {
  describe('on compile', () => {
    it('single path', () => {
      expect(compiler.compile('testElement', {
        paths: ['bar'],
        template: '"test-template-string"'
      })).toMatchSnapshot();
    })
    it('multiple paths', () => {
      expect(compiler.compile('testElement', {
        paths: ['bar', 'foo'],
        template: '"test-template-string"'
      })).toMatchSnapshot();
    })
  })
})
