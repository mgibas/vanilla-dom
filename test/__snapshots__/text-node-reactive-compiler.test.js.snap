// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text-node-reactive-compiler on compile multiple paths 1`] = `
"_reactivePaths['bar'] = _reactivePaths['bar'] || [];
_reactivePaths['bar'].push({
  type: 'text',
  element: testElement,
  template: state => 'test-template-string'
});
_reactivePaths['foo'] = _reactivePaths['foo'] || [];
_reactivePaths['foo'].push({
  type: 'text',
  element: testElement,
  template: state => 'test-template-string'
});"
`;

exports[`text-node-reactive-compiler on compile single path 1`] = `
"_reactivePaths['bar'] = _reactivePaths['bar'] || [];
_reactivePaths['bar'].push({
  type: 'text',
  element: testElement,
  template: state => 'test-template-string'
});"
`;
